src/components/UserSelector.tsx

```diff
@@ -1,19 +1,48 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useContext, useState, useEffect } from 'react';
+import { User } from '../types/User';
+import { UserContext } from './UserContext';
+
+type Props = {
+  selectedUser: User | null;
+  setSelectedUser: (user: User) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  setSelectedUser,
+}) => {
+  const users = useContext(UserContext);
+  const [isOpenDropdown, setIsOpenDropdown] = useState(false);
+
+  useEffect(() => {
+    const handlePageClick = () => {
+      if (isOpenDropdown) {
+        setIsOpenDropdown(false);
+      }
+    };
+
+    document.addEventListener('click', handlePageClick);
```

Comment:
why do you handle click by eventListener?
it makes no sense in react app, add onClick with handler

Reply:
didn't fix it

Reply:
i want to do it like this

Reply:
I want to do it like this in this app

Reply:
Remove use Effect with eventListener - it brokes yur dropdown